src/linear_algebra/invariant_submodule.lean

```diff
@@ -0,0 +1,19 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import linear_algebra.basic
+
+/-!
+# Invariant submodules
+
+In this file, we define and prove some basic results on invaraint submodules.
+-/
+
+/-- `U` is `T` invariant: `∀ u : V`, if `u ∈ U` then `T u ∈ U`-/
+def submodule.invariant {V K : Type*} [semiring K] [add_comm_monoid V] [module K V]
+  (U : submodule K V) (T : V →ₗ[K] V) : Prop := U ≤ U.comap T
+lemma submodule.invariant_iff {V K : Type*} [semiring K] [add_comm_monoid V] [module K V]
+  (U : submodule K V) (T : V →ₗ[K] V) : submodule.invariant U T ↔ T '' U ⊆ U :=
+by simp only [set.image_subset_iff]; refl
```

Comment:
This should be stated in terms of `submodule.map`

Reply:
what do you mean?

Reply:
Sorry I should have been more explicit. My instinct here is that we need (at least) the three following ways to express `submodule.invariant`:
- a version in terms of `submodule.comap` (which you have as the definition) because it is definitionally equal (feel free to ask on Zulip if you don't know what this means) to the natural `∀ x ∈ U, T x ∈ U`
- a version in terms of `submodule.map` because when you're doing computations on submodules `map` is sometimes more natural than `comap`
- a version for the less-used `set.maps_to` because it has a really nice API and should be used more

If the `set.image` version you have here is really useful we can keep it too, but in general it's better to stay in the world of submodules if you can, whereas your lemma coerces everything back into `set`.

Does that make sense?

Reply:
I think I did all this. Check it out in `linear_algebra/invariant_submodule`.